### PR TITLE
Use interfaces when testing conditions so we don't need versioned methods

### PR DESCRIPTION
**Pull request by dprotaso**
### Downstream PRs

- [x] knative/serving - https://github.com/knative/serving/pull/6985
- [x] knative/eventing - noop (no usage)
- [x] knative/eventing-contrib - noop (no usage)
- [x] knative/sample-controller - noop (no usage)
- [x] knative/sample-source - noop (no usage)

### CODE REVIEW

**Comment by dprotaso** [comment]
hmm
/test pull-knative-pkg-unit-tests

File: apis/testing/conditions.go [hunk 1]
```diff
@@ -20,14 +20,12 @@ import (
...
 	"knative.dev/pkg/apis"
-	duckv1b1 "knative.dev/pkg/apis/duck/v1beta1"
 	"knative.dev/pkg/test"
 )
 
 // CheckCondition checks if condition `c` on `cc` has value `cs`.
-// DEPRECATED: Use versioned test helper
-func CheckCondition(s *duckv1b1.Status, c apis.ConditionType, cs corev1.ConditionStatus) error {
-	cond := s.GetCondition(c)
+func CheckCondition(a apis.ConditionAccessor, c apis.ConditionType, cs corev1.ConditionStatus) error {
```
Comment (n3wscott) [inline apis/testing/conditions.go]: Why not use the condtionManager interface?

Comment (dprotaso) [reply apis/testing/conditions.go]: That interface is huge for what these helpers need

Comment (dprotaso) [reply apis/testing/conditions.go]: secondly we're passing `duck.Status` types here

Comment (n3wscott) [reply apis/testing/conditions.go]: You might also want gettoplevel in the new interface

**Comment by dprotaso** [comment]
`knative/eventing` doesn't use these helpers so that's done

**Comment by dprotaso** [comment]
`knative/eventing-contrib` doesn't use these helpers so that's done

**Comment by dprotaso** [comment]
`knative/sample-controller` doesn't use these helpers so that's done

**Comment by dprotaso** [comment]
`knative/sample-source` doesn't use these helpers so that's done

File: apis/testing/conditions.go (hunk 1 above)
Comment (dprotaso) [reply apis/testing/conditions.go]: `duck.Status` doesn't have that method so don't want to add it

**Comment by dprotaso** [comment]
@n3wscott PTAL again

**Pull request merged**